test/unit/YawPitchControl/ScreenRotationAngle.spec.js

```diff
@@ -0,0 +1,195 @@
+import TestHeler from "./testHelper";
+import {window} from "../../../src/YawPitchControl/browser";
+import ScreenRotationAngle from "../../../src/YawPitchControl/ScreenRotationAngle";
+import ScreenRotationAngleInjector from "inject-loader!../../../src/YawPitchControl/ScreenRotationAngle";
+
+describe("ScreenRotationAngle", function() {
+	describe("#constructor", function() {
+		it("Instance", () => {
+			// Given
+			// When
+			this.inst = new ScreenRotationAngle();
+
+			// Then
+			expect(this.inst).to.be.exist;
+
+			// cleanup for later test
+			this.inst.unref();
+		});
+
+		it("should return singleton instance.", () => {
+			// Given
+			// When
+			const sr1 = new ScreenRotationAngle();
+			const sr2 = new ScreenRotationAngle();
+
+			// Then
+			expect(sr1).to.deep.equal(sr2);
+
+			// cleanup
+			sr1.unref();
+			sr2.unref();
+		});
+	});
+
+	describe("#getRadian", function() {
+		it("should return 0 when first", () => {
+			// Given
+			// When
+			const sr = new ScreenRotationAngle();
+
+			// Then
+			expect(sr.getRadian()).to.equal(0);
+
+			// cleanup for later test
+			sr.unref();
+		});
+
+		it("should reflect a deviceorientation change", done => {
+			// Given
+			const sr = new ScreenRotationAngle();
+
+			// When
+			// Dispatch 'deviceorientation' event.
+			const event = TestHeler.createDeviceOrientationEvent(0, 0, 90);
+			window.dispatchEvent(event);
+
+			// Then
+			TestHeler.once(window, "deviceorientation", e => {
+				expect(sr.getRadian()).to.equal(Math.PI / 2);
+
+				// cleanup for later test
+				sr.unref();
+				done();
+			});
+		});
+	});
+
+	describe("#orientationchange", () => {
+		it("should returns rotated angles when screen is rotated.", () => {
+			// Given
+			const win = {
+				screen: {
+					orientation: {
+						angle: 90
+					}
+				}
+			};
+
+			const ScreenRotationAngle90 = ScreenRotationAngleInjector({"./browser": {
+				"window": window,
+				"screen": win.screen
+			}}).default;
+
+			const sr = new ScreenRotationAngle90();
+
+			// When
+			// Dispatch 'orientationchange' event.
+			// const event = TestHeler.createOrientationChangeEvent();
+			// window.dispatchEvent(event);
+
+			// // Then
+			// TestHeler.once(window, "orientationchange", e => {
+			// 	expect(sr.getRadian()).to.equal(0);
+
+			// 	// cleanup for later test
+			// 	sr.unref();
+			// 	done();
+			// });
+			/**
+			 * It's impossible to dispatch orientationchange event.
+			 * So here we go with calling private method.
+			 */
+			// sr._onOrientation
+			sr._onOrientationChange()
```

Comment:
FYI, you can dispatch "orientationchange" event promatically like below.
```js 
var evt = document.createEvent("HTMLEvents");
evt.initEvent("orientationchange", true, true);
window.dispatchEvent(evt);
```

I got this from here. https://gist.github.com/jlubean/3287069

Reply:
It was mistake . I'll apply it.